cmd/snap-confine/snap-confine-args.c

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "snap-confine-args.h"
+
+#include <string.h>
+
+#include "utils.h"
+
+struct sc_args {
+	// The security tag that the application is intended to run with
+	char *security_tag;
+	// The executable that should be invoked
+	char *executable;
+
+	// Flag indicating that --version was passed on command line.
+	bool is_version_query;
+	// Flag indicating that --classic was passed on command line.
+	bool is_classic_confinement;
+};
+
+struct sc_args *sc_nonfatal_parse_args(int *argcp, char ***argvp,
+				       struct sc_error **errorp)
+{
+	struct sc_args *args = NULL;
+	struct sc_error *err = NULL;
+
+	if (argcp == NULL || argvp == NULL) {
+		err = sc_error_init(SC_ARGS_DOMAIN, 0,
+				    "cannot parse arguments, argc or argv are NULL");
```

Comment:
I'll answer the last comment first, no reason, I need to review what the common way is and just apply that everywhere.